src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -4014,7 +4014,8 @@ static ssize_t icap_write_rp(struct file *filp, const char __user *data,
 		icap->rp_mgmt_bin_len = section->m_sectionSize;
 	}
 
-	if (XDEV(xdev)->priv.sched_bin) {
+	if (XDEV(xdev)->priv.sched_bin
+		&& !(XDEV(xdev)->priv.flags & XOCL_DSAFLAG_DYNAMIC_IP)) {
```

Comment:
Why don't we remove priv.sched_bin defined in devices.h for u250?  And when we need force sched_bin for one of dynamic platform, we can still do it.